x/mongo/driver/batch_cursor.go

```diff
@@ -94,6 +99,27 @@ func NewCursorResponse(response bsoncore.Document, server Server, desc descripti
 			}
 		}
 	}
+
+	// If the deployment is behind a load balancer and the cursor has a non-zero ID, pin the cursor to a connection and
+	// use the same connection to execute getMore and killCursors commands.
+	if curresp.Desc.ServerID != nil && curresp.ID != 0 {
+		// Cache the server as an ErrorProcessor to use when constructing deployments for cursor commands.
+		ep, ok := curresp.Server.(ErrorProcessor)
+		if !ok {
+			return CursorResponse{}, fmt.Errorf("expected Server used to establish a cursor to implement ErrorProcessor, but got %T", curresp.Server)
+		}
+		curresp.ErrorProcessor = ep
+
+		refConn, ok := info.Connection.(PinnedConnection)
+		if !ok {
+			return CursorResponse{}, fmt.Errorf("expected Connection used to establish a cursor to implement ReferencedConnection, but got %T", info.Connection)
```

Comment:
to implement "PinnedConnection" (but, perhaps to my above point, if all we need is for it to implement `ReferencedConnection`, then ...

Reply:
Ah, my bad. `ReferencedConnection` is from an older implementation. Fixed.